transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionRepository.java

```diff
@@ -0,0 +1,209 @@
+                    .setGlobalPrepare(this::createTable)
+                    .setPrepare(tag -> prepareStatements())
+                    .setEventHandler(TransactionEvent.TransactionStarted.class,
+                            e -> insertUserTransactions(e.getItemId(), e.getTransaction()))
+                    .build();
+        }
+
+        @Override
+        public PSequence<AggregateEventTag<TransactionEvent>> aggregateTags() {
+            return TransactionEvent.TAG.allTags();
+        }
+
+        private CompletionStage<Done> createTable() {
+            return session.executeCreateTable(
+                    "CREATE TABLE IF NOT EXISTS userTransactions (" +
+                            "userId UUID, " +
+                            "itemId timeuuid, " +
+                            "creatorId UUID, " +
+                            "winnerId UUID, " +
+                            "itemTitle text, " +
+                            "currencyId text, " +
+                            "itemPrice int, " +
+                            "status text, " +
+                            "PRIMARY KEY (userId, status, itemId)" +
+                            ") " +
+                            "WITH CLUSTERING ORDER BY (status ASC, itemId DESC)"
+            );
+        }
+
+        private CompletionStage<Done> prepareStatements() {
+            return doAll(
+                    session.underlying()
+                            .thenAccept(s -> registerCodec(s, new EnumNameCodec<>(TransactionInfoStatus.class)))
+                            .thenApply(x -> Done.getInstance()),
+                    prepareInsertTransactionStatement()
+            );
+        }
+
+        private CompletionStage<Done> prepareInsertTransactionStatement() {
+            return session.
+                    prepare("INSERT INTO userTransactions(" +
+                            "userId, " +
+                            "itemId, " +
+                            "creatorId, " +
+                            "winnerId, " +
+                            "itemTitle, " +
+                            "currencyId, " +
+                            "itemPrice, " +
+                            "status" +
+                            ") VALUES (" +
+                            "?, " + // userId
+                            "?, " + // itemId
+                            "?, " + // creatorId
+                            "?, " + // winnerId
+                            "?, " + // itemTitle
+                            "?, " + // currencyId
+                            "?, " + // itemPrice
+                            "?" +   // status
+                            ")"
+                    )
+                    .thenApply(accept(s -> insertUserTransactionsStatement = s));
+        }
+
+        private CompletionStage<List<BoundStatement>> insertUserTransactions(UUID itemId, Transaction transaction) {
+            return CassandraReadSide.completedStatements(
+                    Arrays.asList(
```

Comment:
This could make use of `CassandraReadSideUtils.completedStatements` to avoid the need to call `Arrays.asList`. This is exactly what it's there for! 😄